src/ocaml/typing/cmt_format.ml

```diff
@@ -467,21 +467,23 @@ let read_cmi filename =
     | Some cmi, _ -> cmi
 
 let saved_types = ref []
-let value_deps = ref []
+let uids_deps : (dependency_kind * Uid.t * Uid.t) list ref = ref []
 
 let clear () =
   saved_types := [];
-  value_deps := []
+  uids_deps := []
 
 let add_saved_type b = saved_types := b :: !saved_types
 let get_saved_types () = !saved_types
 let set_saved_types l = saved_types := l
 
-(*let record_value_dependency vd1 vd2 =
-  if vd1.Types.val_loc <> vd2.Types.val_loc then
-    value_deps := (vd1, vd2) :: !value_deps*)
+(*
+let record_declaration_dependency (rk, uid1, uid2) =
+  if not (Uid.equal uid1 uid2) then
+    uids_deps := (rk, uid1, uid2) :: !uids_deps
+*)
 
-let record_value_dependency _vd1 _vd2 = ()
+let record_declaration_dependency (_rk, _uid1, _uid2) = ()
```

Comment:
@voodoos I don't understand why `record_value_dependency`'s implementation was made to just be `()`. Do you know what's going on here / whether this is correct?

Reply:
I am not sure where that comes from: it is not the case in my initial pr: https://github.com/oxcaml/merlin/pull/140/files#diff-134bb073775f627e1993cd57cf442f3736ebd55ad1add766f032f856750f5c89R480

We definitively want to record these for the active buffer. (and we probably need to check how this interact with the typer cache, but that can be fixed later)

Reply:
Thanks, I'll push a commit to change it.